### ISSUE TITLE
Easily Visible Mobs (highlighting)

### ISSUE THREAD

**Issue by leviem1** [garbagemule/MobArena#629]
# Feature request

**Short description**
I have many younger players in my server, and they are becoming frustrated while searching for mobs in larger arenas, especially spiders and wolves that will turn passive if left alone. One of my players asked for the enemies to be "highlighted" near the end of a round.

**Implementation details**
A configurable option to enable "highlighting" (much how spectral arrows work) for mobs when there are a certain number of enemies left or total time spent per wave.

Example:
```
highlight-after-seconds: 120
highlight-max-mobs: 3
```
This will allow the players to spend less time searching and more time playing.

**Additional info**

Side note: Thought your use of html comments was super clever for the issue templates. 😄

**Comment by garbagemule** [comment]
It's possible right now to give monsters potion effects, but they will have the effects as soon as the wave spawns.

I'm always very reluctant to implement hyper-specific features like this one, mostly because this is the only time it has been suggested, and I fear the implementation will be too complex to really justify the effort of trying to coerce the code base into something that will allow it.

That doesn't mean the idea is bad - I just don't think it's general enough.

What I think we _could_ do with is the idea of monster "phases". This very quickly becomes Mobs Rework territory, but the idea of giving monsters configurable "auras" or "effects" or "phases" that can be triggered by various conditions, e.g. health below 20 %, time alive above 30 seconds, etc., is much more general and will allow for so much more than a hand-holding workaround for a hard-to-navigate arena design - arenas should of course still be playable even if they are hard to navigate.

There is a small issue with the wording in the feature request as well, and that's "... near the end of a round". By round I guess you mean wave. Waves don't have well-defined "ends", and that makes it super difficult to do any sort of "reverse countdown" (people have suggested "countdown to next wave" functionality previously, but that doesn't work with `clear-wave-before-next: true`). The example config-file excerpt does count "from the start of the wave", but there is definitely a lot of edge case issues that will either make this sort of functionality a bit difficult to reason about.

TL;DR: I think highlighting mobs from the get-go using the current `effects` node is the best workaround for now, and while I do like the more general idea of auras/phases/triggers, it's just really hard to pull off with the current code base.

**Comment by leviem1** [comment]
Agreed "end of the round" wasn't proper wording. As you probably guessed from my example it's more of a "timeout"/countdown from the start of the wave. The true root of the issue is that there are many mobs that will stop targeting the player at a certain distance. This wouldn't matter much except that any arenas with grass in them or that exceed the mob's target distance blocks turn into a really boring and difficult game of hide-and-seek. Arenas shouldn't need bare floors and be restricted by size to function well. I'm literally putting a glass ring around a vanilla village.

I imagine the solution you mentioned with conditional event triggers would be just as effective at solving this, and many other problems though.
